include/libtorrent/peer_list.hpp

```diff
@@ -183,6 +183,10 @@ namespace libtorrent
 
 		std::pair<iterator, iterator> find_peers(address const& a)
 		{
+#if TORRENT_USE_I2P
+			if (a.to_v4() == address())
```

Comment:
I think I just left a comment here. but I think you'd have to drop the to_v4() call. If a is IPv6, it would throw.